core/src/avm2/e4x.rs

```diff
@@ -244,6 +244,57 @@ impl<'gc> E4XNode<'gc> {
         node
     }
 
+    /// Returns the amount of children in this node if this node is of Element kind, otherwise returns [None].
+    pub fn length(&self) -> Option<usize> {
+        let E4XNodeKind::Element { children, .. } = &*self.kind() else {
```

Comment:
nit: I'd say do it the other way around?
```
it let E4XNodeKind::Element { children, .. } = &*self.kind() {
    Some(children.len())
} else {
    None
}
```

Reply:
Can cleanup in follow up PR.